sky/shell/ui/flutter_font_selector.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_SHELL_UI_FLUTTER_FONT_SELECTOR_H_
+#define SKY_SHELL_UI_FLUTTER_FONT_SELECTOR_H_
+
+#include <vector>
+
+#include "sky/engine/platform/fonts/FontCacheKey.h"
+#include "sky/engine/platform/fonts/FontSelector.h"
+#include "sky/engine/platform/fonts/SimpleFontData.h"
+
+namespace mojo {
+namespace asset_bundle {
+class ZipAssetBundle;
+}
+}
+
+namespace sky {
+namespace shell {
+
+// A FontSelector implementation that resolves custon font names to assets
+// loaded from the FLX.
+class FlutterFontSelector : public blink::FontSelector {
+ public:
+  static void install(mojo::asset_bundle::ZipAssetBundle* zip_asset_bundle);
+
+  PassRefPtr<blink::FontData> getFontData(
+      const blink::FontDescription& font_description,
+      const AtomicString& family_name) override;
+
+  void willUseFontData(const blink::FontDescription& font_description,
+                       const AtomicString& family,
+                       UChar32 character) override;
+
+  unsigned version() const override;
+
+  void fontCacheInvalidated() override;
+
+ private:
+  // A Skia typeface along with a buffer holding the raw typeface asset data.
+  struct TypefaceAsset {
+    RefPtr<SkTypeface> typeface;
+    std::vector<uint8_t> data;
+  };
+
+  FlutterFontSelector(mojo::asset_bundle::ZipAssetBundle* zip_asset_bundle);
+
+  void parseFontManifest();
+  SkTypeface* getTypefaceAsset(const AtomicString& family_name);
+
+  mojo::asset_bundle::ZipAssetBundle* zip_asset_bundle_;
```

Comment:
I see we have the same lifetime problem here.

We probably need some sort of refcounting scheme to keep the assent bundle alive.